src/streamlink/plugins/atresplayer.py

```diff
@@ -23,30 +22,18 @@
     r"https?://(?:www\.)?atresplayer\.com/",
 ))
 class AtresPlayer(Plugin):
+    _channels_api_url = "https://api.atresplayer.com/client/v1/info/channels"
+    _player_api_url = "https://api.atresplayer.com/player/v1/live/{channel_id}"
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.url = update_scheme("https://", f"{self.url.rstrip('/')}/")
 
     def _get_streams(self):
-        path = urlparse(self.url).path
-        api_url = self.session.http.get(self.url, schema=validate.Schema(
-            re.compile(r"""window.__PRELOADED_STATE__\s*=\s*({.*?});""", re.DOTALL),
-            validate.none_or_all(
-                validate.get(1),
-                validate.parse_json(),
-                {"links": {path: {"href": validate.url()}}},
-                validate.get(("links", path, "href")),
-            ),
-        ))
-        if not api_url:
-            return
-        log.debug(f"API URL: {api_url}")
-
-        player_api_url = self.session.http.get(api_url, schema=validate.Schema(
-            validate.parse_json(),
-            {"urlVideo": validate.url()},
-            validate.get("urlVideo"),
-        ))
+        channel_path = re.search(r"(\/\w+\/)$", self.url).group()
+        json = self.session.http.get(self._channels_api_url).json()
+        channel_id = [x["id"] for x in json if x["link"]["url"] == channel_path][0]
```

Comment:
This needs to be rewritten using a validation schema. Otherwise, it'll spit out all kinds of random errors that are hard to understand for users of this plugin.

```suggestion
        channel_id = self.session.http.get(self._channels_api_url, schema=validate.Schema(
            validate.parse_json(),
            [{
                "id": str,
                "link": {"url": str},
            }],
            validate.filter(lambda item: item["link"]["url"] == channel_path),
            validate.get((0, "id")),
        ))
```

Reply:
Thanks, I wanted to do it properly with schemas at first, but I couldn't find any good docs about validation schemas. Is there any?